### ISSUE TITLE
[Feature request] Automatic legend extraction for compound plots with a shared legend

### ISSUE THREAD

**Issue by apcamargo** [wilkelab/cowplot#140]
If the user wants to draw a compound plot with a shared legend, it needs to manually extracted the legend and add it to the compound figure. A nicer interface would be to add an option to `plot_grid` to request a shared legend ([akin to the `ggarrange` function of `ggpubr`](https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html)).

**Comment by clauswilke** [comment]
Is there anything wrong with using `ggpubr::ggarrange()`? It's just a wrapper around `plot_grid()` anyways.

**Comment by apcamargo** [comment]
No, no problem at all. I didn't check the code of `ggarrange`, so I didn't know it uses `plot_grid`.
